pyro/poutine/trace_messenger.py

```diff
@@ -148,9 +146,7 @@ def __call__(self, *args, **kwargs):
             except (ValueError, RuntimeError):
                 exc_type, exc_value, traceback = sys.exc_info()
                 shapes = self.msngr.trace.format_shapes()
-                six.reraise(exc_type,
-                            exc_type(u"{}\n{}".format(exc_value, shapes)),
-                            traceback)
+                raise exc_type(u"{}\n{}".format(exc_value, shapes)).with_traceback(traceback)
```

Comment:
Does this get rid of the problem of duplicate stack trace that @fritzo mentioned?